From c4b528fa29f688f20b9127c6550d520a96d991c3 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Tue, 16 Feb 2016 13:52:34 +0200
Subject: [PATCH 4/8] arm: imx7d: cl-som-imx7: add support for status LED

Add support for the on-board status LED. Swith the LED on
at the end of board_init() routine.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 board/compulab/cl_som_imx7/cl_som_imx7.c | 23 +++++++++++++++++++++++
 include/configs/cl_som_imx7.h            | 10 ++++++++++
 2 files changed, 33 insertions(+)

diff --git a/board/compulab/cl_som_imx7/cl_som_imx7.c b/board/compulab/cl_som_imx7/cl_som_imx7.c
index 0cd3a05..0632931 100644
--- a/board/compulab/cl_som_imx7/cl_som_imx7.c
+++ b/board/compulab/cl_som_imx7/cl_som_imx7.c
@@ -103,6 +103,27 @@ static void setup_iomux_otg(void)
 	imx_iomux_v3_setup_multiple_pads(usb_otg1_pads, ARRAY_SIZE(usb_otg1_pads));
 }
 
+#if defined(CONFIG_STATUS_LED) && defined(STATUS_LED_BOOT)
+static iomux_v3_cfg_t const status_led_pads[] = {
+	MX7D_PAD_SAI1_TX_SYNC__GPIO6_IO14 | MUX_PAD_CTRL(GPIO_PAD_CTRL),
+};
+
+static void setup_iomux_led(void)
+{
+	imx_iomux_v3_setup_multiple_pads(status_led_pads, ARRAY_SIZE(status_led_pads));
+}
+
+
+static void setup_led(void)
+{
+	setup_iomux_led();
+	status_led_init();
+	status_led_set(STATUS_LED_BOOT, STATUS_LED_ON);
+}
+#else
+static void setup_led(void) {}
+#endif
+
 #ifdef CONFIG_FSL_ESDHC
 
 #define USDHC_PAD_CTRL		(PAD_CTL_DSE_3P3V_32OHM | PAD_CTL_SRE_SLOW | \
@@ -437,6 +458,8 @@ int board_init(void)
 
 	board_spi_init();
 
+	setup_led();
+
 	return 0;
 }
 
diff --git a/include/configs/cl_som_imx7.h b/include/configs/cl_som_imx7.h
index 288ccbc..acf9f6b 100644
--- a/include/configs/cl_som_imx7.h
+++ b/include/configs/cl_som_imx7.h
@@ -226,4 +226,14 @@
 
 #define CONFIG_IMX_THERMAL
 
+/* Status LED */
+#define CONFIG_STATUS_LED
+#define CONFIG_GPIO_LED
+#define CONFIG_BOARD_SPECIFIC_LED
+#define STATUS_LED_BIT			IMX_GPIO_NR(6, 14)
+#define STATUS_LED_STATE		STATUS_LED_OFF
+#define STATUS_LED_PERIOD		(CONFIG_SYS_HZ / 2)
+#define STATUS_LED_BOOT			0
+#define CONFIG_GPIO_LED_INVERTED_TABLE	{STATUS_LED_BOOT,}
+
 #endif	/* __CONFIG_H */
-- 
1.9.1

